MekHQ/src/mekhq/campaign/mission/BotForce.java

```diff
@@ -194,9 +195,7 @@ public void writeToXml(PrintWriter pw1, int indent) {
 
         pw1.println(MekHqXmlUtil.indentStr(indent+1) + "<entities>");
         for (Entity en : entityList) {
-            if (en != null) {
-                pw1.println(AtBScenario.writeEntityWithCrewToXmlString(en, indent + 2, entityList));
-            }
+            pw1.println(AtBScenario.writeEntityWithCrewToXmlString(en, indent + 2, entityList));
```

Comment:
The rest of the PR seems to be about protecting vs NPEs, why remove the null check here?

Reply:
There should be no reasonable way to get a null here anymore as we are filtering the collections in the constructor and in the setter. If someone adds a null item via the getter, I think that should error and then we figure out why that happened, as that seems an error worth catching.

Reply:
Perhaps then a better way would be to log the presence of a null entity and then move on. An error here would, I believe, prevent the game from being saved.

Reply:
Ok, I'll add error logging here.